src/pybind/mgr/iostat/module.py

```diff
@@ -0,0 +1,80 @@
+
+from mgr_module import MgrModule
+
+
+class Module(MgrModule):
+    COMMANDS = [
+        {
+            "cmd": "iostat",
+            "desc": "Get IO rates",
+            "perm": "r",
+            "poll": "true"
+        },
+        {
+            "cmd": "iostat self-test",
+            "desc": "Run a self test the iostat module",
+            "perm": "r"
+        }
+    ]
+
+
+    def __init__(self, *args, **kwargs):
+        super(Module, self).__init__(*args, **kwargs)
+
+
+    def handle_command(self, command):
+        rd = 0
+        wr = 0
+        total = 0
+        rd_ops = 0
+        wr_ops = 0
+        total_ops = 0
+        ret = ''
+
+        if command['prefix'] == 'iostat':
+            r = self.get('io_rate')
+
+            stamp_delta = float(r['pg_stats_delta']['stamp_delta'])
+            if (stamp_delta > 0):
+                rd = int(r['pg_stats_delta']['stat_sum']['num_read_kb']) / stamp_delta
+                wr = int(r['pg_stats_delta']['stat_sum']['num_write_kb']) / stamp_delta
+                # The values are in kB, but to_pretty_si() requires them to be in bytes
+                rd = int(rd) << 10
+                wr = int(wr) << 10
+                total = rd + wr
+
+                rd_ops = int(r['pg_stats_delta']['stat_sum']['num_read']) / stamp_delta
+                wr_ops = int(r['pg_stats_delta']['stat_sum']['num_write']) / stamp_delta
+                total_ops = rd_ops + wr_ops
+
+            if 'width' in command:
+                width = command['width']
```

Comment:
i'd think the backend should be responsible to render the output to accommodate the needs of client. how about reply, for example, a json to the client, and let client to to the formatting?

Reply:
this breaks the test.sh http://pulpito.ceph.com/kchai-2018-04-16_09:06:16-rados-wip-kefu-testing-2018-04-16-1515-distro-basic-smithi/

as the log line now looks like
```
2018-04-16 11:07:08.015 7f2cd768b700  0 log_channel(audit) log [DBG] : from='client.8920 172.21.15.10:0/2640242070' entity='client.admin' cmd=[{"width": 80, "prefix": "version"}]: dispatch
``` 

with this change.

Reply:
We discussed the idea of having the client do formatting in the original iostat PR -- the trouble with that is that the client side doesn't know enough about the data (units, which columns are most important, etc) to format it, unless we invent quite a complicated language for describing tables.